src/test/java/nl/ordina/jobcrawler/controller/SkillControllerTest.java

```diff
@@ -26,23 +40,181 @@
+                        .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+
+                        .andExpect(content().string(asJsonString(skill)));
+
+            }
+
+            @Test
+            @DisplayName("GET - Not Found")
+            void testGetSkillByIDNotFound() throws Exception {
+
+                doReturn(Optional.empty()).when(skillService).findById(any(UUID.class));
+
+                mockMvc.perform(get("/getByID/{id}", UUID.randomUUID()))
+                        .andExpect(status().isNotFound());
+            }
+        }
+
+        @Nested
+        @DisplayName("PUT")
+        class put {
+            @Test
+            @DisplayName("PUT - Not Found")
+            void testUpdateSkillNotFound() throws Exception {
+                Skill skill = SkillFactory.create("skill1");
+                Skill newSkill = SkillFactory.create("skill2");
+
+                doReturn(Optional.empty()).when(skillService).findById(any(UUID.class));
+
+                mockMvc.perform(put("/skills/{id}", UUID.randomUUID())
+                        .content(asJsonString(newSkill))
+                        .contentType(MediaType.APPLICATION_JSON))
+
+                        .andExpect(status().isNotFound());
+            }
+
+            // TODO: For later increment
+            @Disabled
+            @Test
+            @DisplayName("PUT - Conflict")
+            void testUpdateSkillConflict() throws Exception {
+                Skill skill = SkillFactory.create("skill1");
+                Skill newSkill = SkillFactory.create("skill2");
+
+//                doReturn(Optional.empty()).when(skillService).replace(skill.getId(), newSkill);
+
+                mockMvc.perform(put("/skills/{id}", skill.getId())
+                        .content(asJsonString(newSkill))
+                        .contentType(MediaType.APPLICATION_JSON)
+
+                        .header(HttpHeaders.IF_MATCH, 12345))
+                        .andExpect(status().isConflict());
+            }
+
+            @Test
+            @DisplayName("PUT - OK")
+            void testUpdateSkillSuccess() throws Exception {
+                Skill skill = SkillFactory.create("skill");
+                Skill newSkill = new Skill("new skill");
+
+                Skill updatedSkill = SkillFactory.create(skill.getId(), newSkill.getName());
+
+                doReturn(Optional.of(skill)).when(skillService).findById(skill.getId());
+                doReturn(updatedSkill).when(skillService).update(skill.getId(), newSkill);
+
+                mockMvc.perform(put("/skills/{id}", skill.getId())
+
+                        .content(asJsonString(newSkill))
+                        .contentType(MediaType.APPLICATION_JSON))
+
+                        .andExpect(status().isOk())
+                        .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                        .andExpect(content().string(asJsonString(updatedSkill)));
+            }
+        }
```

Comment:
This test does not pass even though it works when running the applications and testing. 
Running the test gives this error
```
java.lang.AssertionError: Content type not set

	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:37)
	at org.springframework.test.util.AssertionErrors.assertTrue(AssertionErrors.java:70)
	at org.springframework.test.util.AssertionErrors.assertNotNull(AssertionErrors.java:106)
	at org.springframework.test.web.servlet.result.ContentResultMatchers.lambda$contentType$0(ContentResultMatchers.java:83)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:196)
	at nl.ordina.jobcrawler.controller.SkillControllerTest$id$put.testUpdateSkillSuccess(SkillControllerTest.java:183)
...
```
Tried to search the issue but did not come to a working solution